posthog-core/src/index.ts

```diff
@@ -456,6 +457,12 @@ export abstract class PostHogCoreStateless {
   }
 
   flush(callback?: (err?: any, data?: any) => void): void {
+    if (this.isFlushing) {
+      console.log('[PostHog] The Queue is already flushing.')
+      return
```

Comment:
Hmmmm the callback is important here otherwise the async calls never fire... 
If we add the callback it becomes tricky as it means the flushAsync will resolve immediately which is not really true 🤔 

We could do a promise-hold approach perhaps instead which would give us a way of tracking the inflight flush?

Reply:
This is made more important by the fact it affects Node as well

Reply:
yeah should have called `callback?.()` here, and pushed a change just so that the `flushAsync` works as it should.

> flushAsync will resolve immediately which is not really true 🤔

yep but it's fine I think, I'd consider that a best effort, the `_flushTimer` will get triggered again in the next cycle.

The only issue I see is the `shutdownAsync`, and `flushAsync` will return right away but even with that, it's not guaranteed that the whole Queue will be flushed even after calling `flushAsync` twice, so if the flush was on going during the `shutdownAsync` call, I'd consider that ok and send the rest next time the SDK is up and running again.
If the Queue has to be fully flushed before shutdown sync is completed, even if the current implementation does not work, we'd need to iterate the Queue until it's empty.

Reply:
discussed offline, ideally, the shutdown should not bail if it's already flushing but rather awaits the pending flushes and flushes everything else as well until the very end (iterate the queue), this is important for node, specifically serverless/cloud functions stuff that boots and dies right away, all the pending events has to be flushed otherwise they are lost.